playbooks/common/openshift-cluster/additional_config.yml

```diff
@@ -23,12 +23,8 @@
     when: openshift.common.use_manageiq | bool
   - role: cockpit
     when: not openshift.common.is_atomic and ( deployment_type in ['atomic-enterprise','openshift-enterprise'] ) and
-      (osm_use_cockpit | bool or osm_use_cockpit is undefined )
-  - role: flannel_register
-    when: openshift.common.use_flannel | bool
-
-- name: Cockpit UI configuration
-  hosts: oo_first_master
-  roles:
+      (osm_use_cockpit | bool or osm_use_cockpit is undefined ) and ( deployment_subtype != 'registry' )
```

Comment:
Condition probably should be:
```
osm_use_cockpit | default(false) | bool and deployment_subtype != 'registry'
```
The deployment_subtype could be moved to the default filter, but I think that would be a bit counter-intuitive to read.

That said, I think this is a bit confusing to follow that the setting of deployment_subtype would override the osm_use_cockpit variable in that way.

Would it make sense to possibly deploy the cockpit container by default on all systems as opposed to doing the package based deployment?